src/rollover.py

```diff
@@ -486,24 +477,55 @@ def main_rollover(args):
     return return_value
 
 
-def main_docker_stop(args):
+def run_with_timeout(instance_id, command, timeout):
     """
-    Main entry point for the docker-stop command
+        @param instance_id str
+        @param command str
+        @param timeout int
+        @return bool if command succeeded
     """
-    ec2_client = ec2.EC2Client()
-    info = ec2_client.describe_instances([args.ec2_id])
-    return ssh.stop_all_containers(info[args.ec2_id]['PrivateIpAddress'],
-                                   args.timeout)
 
+    client = boto3.client('ssm')
+    response = client.send_command(
+        InstanceIds = [instance_id],
+        DocumentName = 'AWS-RunShellScript',
+        TimeoutSeconds = max(timeout, 30),
+        Comment = '',
+        Parameters = {
+            'commands': ["#!/bin/bash", command]
+        },
+        OutputS3BucketName = 'clever-test',
+        OutputS3KeyPrefix = 'test-rollover'
+    )
+
+    # Error sending response
+    if response.get('ResponseMetadata', {}).get('HTTPStatusCode') is not 200:
+        print "ERROR sending command:", response
+        return False
+
+    command_id = response.get('Command').get('CommandId')
+
+    # Not sure when/if this actually happens; adding as a safeguard for now.
+    if not command_id:
+        return False
+
+    invocation_response = client.list_command_invocations(CommandId=command_id, InstanceId=instance_id, Details=True)
+    # Wait until command reaches final state
+    while invocation_response.get('Status') in ['Pending', 'InProgress', 'Cancelling']:
+        time.sleep(2)
+        invocation_response = client.list_command_invocations(CommandId=command_id, InstanceId=instance_id, Details=True)
+
+    # To get the command output
+    output = invocation_response.get('CommandInvocations')[0].get('CommandPlugins')[0].get('Output')
```

Comment:
Once I remove debug print statements, no. Part of me wants to keep it just because python won't flip out if I declare an unused variable, but… I won't.